### PR TITLE
Forms: TextArea component 

### LINKED ISSUE #20265: Forms: Textarea component
Implement a Textarea as part of #19855

### PR DESCRIPTION
**Pull request by peterholmberg**
**What this PR does / why we need it**:
Adds a TextArea component

**Which issue(s) this PR fixes**:
Fixes #20265 

**Special notes for your reviewer**:

### CODE REVIEW

File: packages/grafana-ui/src/components/Forms/commonStyles.ts [hunk 1]
```diff
@@ -1,9 +1,35 @@
...
 export const getFocusStyle = (theme: GrafanaTheme) => css`
   &:focus {
     outline: none;
     box-shadow: 0 0 0 2px ${theme.colors.blueLight};
   }
 `;
+
+export const sharedInputStyle = (theme: GrafanaTheme) => {
+  const colors = theme.colors;
+  const backgroundColor = selectThemeVariant({ light: colors.white, dark: colors.gray15 }, theme.type);
```
Comment (dprokop) [inline packages/grafana-ui/src/components/Forms/commonStyles.ts]: Use `formInputBg` variable pls

File: packages/grafana-ui/src/components/Forms/commonStyles.ts [hunk 2]
```diff
@@ -1,9 +1,35 @@
...
+    &:hover {
+      border-color: ${colors.formInputBorder};
+    }
+
+    &:focus {
+      outline: none;
+    }
+
+    &:disabled {
+      background-color: ${selectThemeVariant({ light: colors.gray6, dark: colors.gray10 }, theme.type)};
```
Comment (dprokop) [inline packages/grafana-ui/src/components/Forms/commonStyles.ts]: Use `formInputBgDisabled` variable pls

File: packages/grafana-ui/src/components/Forms/commonStyles.ts [hunk 3]
```diff
@@ -1,9 +1,35 @@
...
+      color: ${selectThemeVariant({ light: colors.gray33, dark: colors.gray70 }, theme.type)};
```
Comment (dprokop) [inline packages/grafana-ui/src/components/Forms/commonStyles.ts]: Introduce new variable for this pls

**Review by dprokop — CHANGES_REQUESTED** [review]

**Comment by dprokop** [comment]
Couple of nits, haven't run that yet. Btw `packages/grafana-ui/src/components/Forms/TextArea/TextArea.mdx` is empty, is that on purpose?

File: packages/grafana-ui/src/components/Forms/commonStyles.ts (hunk 3 above)
Comment (peterholmberg) [reply packages/grafana-ui/src/components/Forms/commonStyles.ts]: :man_facepalming:

**Comment by dprokop** [comment]
@peterholmberg really sorry about late review:( 

So I have tested it and it seems like all working fine! Few notes about improvements I'm thinking about:

- Let's introduce sizing classes, similar to the one's I have introduced to Input in https://github.com/grafana/grafana/pull/20482 (sm, md,lg, auto)
- Let's sync this with input styling updates (focus primarily) introduced in https://github.com/grafana/grafana/pull/20482 . I aim to merge it on monday
- I'm thinking about auto-growing textarea that you can specify axis on which it should grow. I.e. `<Textarea grow="width|height|both" />`. It would disable the resize option on textarea and let it grow  until it reaches container's boundary. This is a nice-to-have thing, could be useful i.e in Http settings for pasting the certificates, or in sidebar panel editor where we probably want the textareas to be as small as possible when rendered. WDYT about this idea @jessover9000 @torkelo

File: packages/grafana-ui/src/components/Forms/Form.story.tsx [hunk 4]
```diff
@@ -82,7 +83,7 @@ export const users = () => {
             label="Path to client cert"
             description="Authentication against LDAP servers requiring client certificates if not required leave empty "
           >
-            <Input
+            <TextArea
               id="clientCert"
               value={''}
               // onChange={e => setPassword(e.currentTarget.value)}
```
Comment (dprokop) [inline packages/grafana-ui/src/components/Forms/Form.story.tsx]:
```suggestion
```

File: packages/grafana-ui/src/components/Forms/TextArea/TextArea.tsx [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+  const borderColor = invalid
+    ? colors.redBase
+    : selectThemeVariant({ light: colors.gray4, dark: colors.gray25 }, theme.type);
+
+  return {
+    textarea: cx(
+      sharedInputStyle(theme),
+      getFocusStyle(theme),
+      css`
+        border: 1px solid ${borderColor};
```
Comment (dprokop) [inline packages/grafana-ui/src/components/Forms/TextArea/TextArea.tsx]: I think we can have this in the shared input styles.

File: packages/grafana-ui/src/components/Forms/TextArea/TextArea.tsx [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+  invalid?: boolean;
+  /** Choose a predefined size */
+  size?: FormInputSize;
+}
+
+const getTextAreaStyle = stylesFactory((theme: GrafanaTheme, invalid = false) => {
+  const colors = theme.colors;
+  const borderColor = invalid
+    ? colors.redBase
+    : selectThemeVariant({ light: colors.gray4, dark: colors.gray25 }, theme.type);
```
Comment (dprokop) [inline packages/grafana-ui/src/components/Forms/TextArea/TextArea.tsx]: formInputBorder variables please :)

**Review by dprokop — APPROVED** [review]
This is looking good now :) Some nits I left comments about

**Pull request merged**